### PR TITLE
Tuned for rootless operation

### PR DESCRIPTION
**Pull request by rkojedzinszky**
This allows running the image rootless.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM php:7.2-apache
-MAINTAINER Pierre Cheynier <pierre.cheynier@gmail.com>
+
+MAINTAINER Richard Kojedzinszky <richard@kojedz.in>
```
Comment (pierrecdn) [inline Dockerfile:3]: I'll still be the maintainer after the merge ;), so adding this makes no sense.

File: Dockerfile [hunk 2]
```diff
@@ -47,30 +48,23 @@ RUN docker-php-ext-configure mysqli --with-mysqli=mysqlnd && \
...
-    -e "s/\['user'\] = 'phpipam'/\['user'\] = getenv(\"MYSQL_ENV_MYSQL_USER\") ?: \"root\"/" \
-    -e "s/\['name'\] = 'phpipam'/\['name'\] = getenv(\"MYSQL_ENV_MYSQL_DB\") ?: \"phpipam\"/" \
-    -e "s/\['pass'\] = 'phpipamadmin'/\['pass'\] = getenv(\"MYSQL_ENV_MYSQL_ROOT_PASSWORD\")/" \
-    -e "s/\['port'\] = 3306;/\['port'\] = 3306;\n\n\$password_file = getenv(\"MYSQL_ENV_MYSQL_PASSWORD_FILE\");\nif(file_exists(\$password_file))\n\$db\['pass'\] = preg_replace(\"\/\\\\s+\/\", \"\", file_get_contents(\$password_file));/" \
-    -e "s/define('BASE', \"\/\")/define('BASE', getenv(\"PHPIPAM_BASE\"))/" \
-    -e "s/\$gmaps_api_key.*/\$gmaps_api_key = getenv(\"GMAPS_API_KEY\") ?: \"\";/" \
-    -e "s/\$gmaps_api_geocode_key.*/\$gmaps_api_geocode_key = getenv(\"GMAPS_API_GEOCODE_KEY\") ?: \"\";/" \
-    ${WEB_REPO}/config.php
+    chown www-data /var/www/html/css/images/logo
+
```
Comment (pierrecdn) [inline Dockerfile:64]: All of this can make sense, but it actually has nothing to do with rootless.

File: Dockerfile [hunk 3]
```diff
@@ -47,30 +48,23 @@ RUN docker-php-ext-configure mysqli --with-mysqli=mysqlnd && \
...
+# Tune for rootless
+RUN sed -i -e 's/^Listen.*/Listen 8080/g' /etc/apache2/ports.conf
+
+USER 33
```
Comment (pierrecdn) [inline Dockerfile:68]: Why user 33?

File: README.md [hunk 4]
```diff
@@ -23,7 +23,7 @@ Here, we store data on the host system under `/my_dir/phpipam` and use a specifi
 ### Phpipam
 
 ```bash
-$ docker run -ti -d -p 80:80 -e MYSQL_ENV_MYSQL_ROOT_PASSWORD=my-secret-pw --name ipam --link phpipam-mysql:mysql pierrecdn/phpipam
+$ docker run -ti -d -p 80:8080 -e IPAM_DATABASE_HOST=mysql -e IPAM_DATABASE_PASS=my-secret-pw --name ipam --link phpipam-mysql:mysql rkojedzinszky/phpipam
 ```
```
Comment (pierrecdn) [inline README.md:27]: Same remark here, doesn't make sense for me to merge this.

File: README.md [hunk 5]
```diff
@@ -83,38 +83,20 @@ And next :
...
+| IPAM_DATABASE_HOST             | localhost     | The host used to reach the MySQL instance                                                                |
+| IPAM_DATABASE_PORT             | 3306          | The port used to reach the MySQL instance                                                                |
+| IPAM_DATABASE_USER             | phpipam       | The user to connect the MySQL instance                                                                   |
+| IPAM_DATABASE_PASS             | phpipamadmin  | The MySQL password. Can be set using the Web UI during the first install                                 |
+| IPAM_DATABASE_NAME             | phpipam       | The name of the MySQL DB to connect to                                                                   |
+| IPAM_BASE                      | /             | The base URI under which phpipam runs. Useful when performing rewrites with a reverse-proxy              |
+| IPAM_GMAPS_API_KEY             | (empty)       | Google Maps API Key, used to display maps of your devices and to find coordinates from an address/ a location of your device |
```
Comment (pierrecdn) [inline README.md:99]: Why this reformating?

**Review by pierrecdn — CHANGES_REQUESTED** [review]
Hi,
While I can agree on the intent, this PR does a lot more, and can't be merged as-is.

**Comment by rkojedzinszky** [comment]
Sorry about the huge changes, I am going to close this PR. I will maintain my rootless version.

**Pull request closed without merging**